pkg/astvalidation/definition_validation.go

```diff
@@ -51,3 +53,20 @@ func (d *DefinitionValidator) Validate(definition *ast.Document, report *operati
 	}
 	return Valid
 }
+
+func IsRootType(nameBytes []byte) bool {
+	length := len(nameBytes)
+	return isQuery(length, nameBytes) || isMutation(length, nameBytes) || isSubscription(length, nameBytes)
+}
+
+func isQuery(length int, b []byte) bool {
+	return length == 5 && b[0] == 'Q' && b[1] == 'u' && b[2] == 'e' && b[3] == 'r' && b[4] == 'y'
```

Comment:
I would suggest to create some constants in the `ast` package like this:
```go
const DefaultQueryTypeName = "Query"
const DefaultMutationTypeName = "Mutation"
const DefaultSubscriptionTypeName = "Subscription"
```

... and use those for comparison. You could also use them to remove some hardcoded values in `ast` package:
https://github.com/TykTechnologies/graphql-go-tools/blob/master/pkg/ast/ast_root_operation_type_definition.go#L54